packages/x-data-grid/src/hooks/utils/useGridComponentRenderer.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import { SxProps } from '@mui/system';
+
+export type ComponentRenderFn<Props, State> = (
+  props: Props,
+  state: State,
+) => React.ReactElement<unknown>;
+
+export type RenderProp<State> =
+  | ComponentRenderFn<React.HTMLAttributes<any>, State>
+  | React.ReactElement;
```

Comment:
Should it be `type RenderProp<Props, State>`?
Otherwise, this won't work:
```tsx
<GridExportTrigger
  render={(props, state) => (
    <button className={
      // ❌ Property 'variant' does not exist on type 'HTMLAttributes<any>'
      props.variant === 'outlined'
        ? 'btn_outlined'
        : 'btn_default'
      }
    />
  )}
/>
```

Reply:
Where would `props.variant` come from though? Wouldn't it make more sense to have the `variant` on the `button` itself?